kernel/packages/unity-interface/loader.ts

```diff
@@ -58,7 +61,12 @@ async function injectRenderer(baseUrl: string, version: string): Promise<LoadRen
         productVersion: '0.1'
       }
 
-      return originalCreateUnityInstance(canvas, config, onProgress)
+      return originalCreateUnityInstance(canvas, config, function (...args) {
+        if (args[0] == 1.0) {
+          trackEvent('unity_started', { version, loading_time: performance.now() - startLoadingTime })
```

Comment:
This event actually tracks the interval for downloading the Unity artifacts, so this should be called `unity_downloading_end` or similar.

Also, we agreed with @sicarul to not calculate the time ourselves but have begin/end events. So we have `unity_downloading_start` as well.

Then we need another event to track the engine initialization time. This is the time between this event and the call to `StartDecentraland()` callback.

Reply:
Yes, please let's split into two events

Reply:
I think even though we _can_ do that, it is easier to have the control of that in our end whenever possible. To avoid race conditions problems like we have right now about changing the sequence order, which leads to break the ETL and metrics

Reply:
Pushed both approaches